### ISSUE TITLE
How to obtain access token (constantly receiving a 400 Bad Request)

### ISSUE THREAD

**Issue by hwo-wd** [ansible-middleware/common#11]
##### SUMMARY
This is less a bug than a mere (noob) question, but:
I'm not able to obtain a proper access token.

TL;DR: When specifying my interactive login credentials for <https://sso.redhat.com> on the oidc endpoint, I get a 400 Bad Request; what's smelly is, that this user is used as `client_id`, I doubt that RedHat would create a client per username inside the `redhat-external` realm; so my question really is: where do I get my credentials for properly authenticating towards <https://sso.redhat.com> such that I can make use of this task (as part of the `keycloak` role)?

##### ISSUE TYPE
 - Question

##### COLLECTION VERSION

```
Collection                     Version
------------------------------ -------
middleware_automation.common   1.1.0
middleware_automation.keycloak 1.2.4
```

##### STEPS TO REPRODUCE

```yaml
- hosts: all
  tasks:
  - name: Retrieve product download using JBoss Network API
    # obtaining access_token: https://github.com/ansible-middleware/common/blob/8186de4987fcb258221bc69f238f8938916b618e/plugins/module_utils/jbossnetwork.py#L53
    middleware_automation.common.product_search:
      client_id: "{{ username_which_I_can_use_to_login_on_sso_redhat_com }}" 
      client_secret: "{{ password_which_I_can_use_to_login_interactively_on_sso_redhat_com }}"
      product_type: DISTRIBUTION
      product_version: "7.6"
      product_category: "core.service.rhsso"
    register: rhn_products
    no_log: false
    delegate_to: localhost
    run_once: true
    tags:
     - test
```

##### EXPECTED RESULTS
Looking at the source code, I tried obtaining an `access_token` using curl directly using the same means as [bossnetwork.py::get_authenticated_session](https://github.com/ansible-middleware/common/blob/8186de4987fcb258221bc69f238f8938916b618e/plugins/module_utils/jbossnetwork.py#L44) does:

```shell
user="user@company.com"
 password="using this very password I'm able to perform *interactive logins* on https://sso.redhat.com/auth/realms/redhat-external/login-actions/authenticate (w/ client_id https://www.redhat.com/wapps/ugc-oidc)"

curl -v \
-d 'client_id=$user' \
-d 'client_secret=$password' \
-d 'grant_type=client_credentials' \
-d 'scope=openid api.iam.service_accounts' \
    'https://sso.redhat.com/auth/realms/redhat-external/protocol/openid-connect/token'
```

Which yields a 400 Bad Request `{"error":"invalid_client","error_description":"Invalid client credentials"}`

##### ACTUAL RESULTS

```
"msg": "Error Retrieving SSO Access Token: 400 Client Error: Bad Request for url: https://sso.redhat.com/auth/realms/redhat-external/protocol/openid-connect/token"
```

Thanks for bearing with me

**Comment by guidograzioli** [comment]
Hello Helmut, thanks for reporting; the documentation is still lacking in that regard, sorry this is a very recent feature.
After you login in the portal to access your hybrid cloud console, go to: https://console.redhat.com/application-services/service-accounts and create a service account. Client ID will be the username and Client secret will be the password

**Comment by hwo-wd** [comment]
Thanks Guido, works!
